New create crawl config user workflow

SuaYoo opened the pull request:
Resolves #364.

### Manual testing
Install new shoelace version with `yarn`, then:
1. Run app with `yarn start`
2. Create new job config
    - Verify URL List flow functions as expected
    - Verify Seeded flow functions as expected
3. Edit newly created job config/crawl template. Verify it can still be edited.
4. Duplicate crawl template. Verify it takes you to new job config flow as URL list.

### Screenshots (in Firefox)
<img width="1139" alt="Screen Shot 2022-11-30 at 6 31 04 PM" src="https://user-images.githubusercontent.com/4672952/204951880-063f1efc-deb5-4080-8143-a7fa9da4ebe2.png">
<img width="900" alt="Screen Shot 2022-11-30 at 6 31 13 PM" src="https://user-images.githubusercontent.com/4672952/204951885-38905f75-fb77-496f-b4f6-57863088d95c.png">
<img width="897" alt="Screen Shot 2022-11-30 at 6 31 23 PM" src="https://user-images.githubusercontent.com/4672952/204951887-2ace5e2d-a1ea-472e-9323-5f69e8addecf.png">
<img width="896" alt="Screen Shot 2022-11-30 at 6 31 34 PM" src="https://user-images.githubusercontent.com/4672952/204951890-76249832-ff37-4786-91e4-fa25d3e61d6c.png">
<img width="903" alt="Screen Shot 2022-11-30 at 6 31 40 PM" src="https://user-images.githubusercontent.com/4672952/204951894-56c64306-9f00-4cdf-b882-605bd68932bd.png">

**Seeded crawl URL:**
<img width="895" alt="Screen Shot 2022-11-30 at 6 31 57 PM" src="https://user-images.githubusercontent.com/4672952/204951896-daf4dda1-5cab-4ba3-84a9-a731388bb75d.png">

**Seeded crawl URL validation:**
<img width="529" alt="Screen Shot 2022-11-30 at 6 32 11 PM" src="https://user-images.githubusercontent.com/4672952/204951897-93329119-20e9-4f63-be46-d4aac17ef104.png">

### Follow-ups
- [x] https://github.com/webrecorder/browsertrix-cloud/pull/392
- [x] https://github.com/webrecorder/browsertrix-cloud/pull/395
- [x] Specify job type (blocked by https://github.com/webrecorder/browsertrix-cloud/issues/363)
- [x] Improve duplicate config (only name and seed URLs copied right now, also blocked by https://github.com/webrecorder/browsertrix-cloud/issues/363)
- [ ] Handle smaller screen sizes: https://github.com/webrecorder/browsertrix-cloud/issues/408

SuaYoo committed:
create new component

SuaYoo committed:
style job type buttons

SuaYoo committed:
update step buttons

SuaYoo committed:
style indicator

SuaYoo committed:
handle navigation

SuaYoo committed:
refactor into tabs

SuaYoo committed:
update crawler state

SuaYoo committed:
update columms

SuaYoo committed:
update progress

SuaYoo committed:
wip

SuaYoo committed:
reposition on resize

SuaYoo committed:
handle field errors

SuaYoo committed:
handle completion

SuaYoo committed:
simplify state

SuaYoo committed:
add form state

SuaYoo committed:
mark as completed

SuaYoo committed:
add url list fields

SuaYoo committed:
update seeded fields

SuaYoo committed:
update crawl scope options

SuaYoo committed:
update linked page options

SuaYoo committed:
add advanced option

SuaYoo committed:
update help text

SuaYoo committed:
remove page limit

SuaYoo committed:
rename browser settings

SuaYoo committed:
update browser settings

SuaYoo committed:
remove trailing bracket

SuaYoo committed:
move job type to new crawl template

SuaYoo committed:
update form state

SuaYoo committed:
update behaviors

SuaYoo committed:
update schedule section

SuaYoo committed:
update cron rendering

SuaYoo committed:
update job information

SuaYoo committed:
udpate validation

SuaYoo committed:
prevent submission from other tab

SuaYoo committed:
prevent submit on enter

SuaYoo committed:
update validation checks

SuaYoo committed:
update layout

SuaYoo committed:
remove unused option

SuaYoo committed:
update time input

SuaYoo committed:
show next scheduled time

SuaYoo committed:
format config

SuaYoo commented:
Blocked by https://github.com/webrecorder/browsertrix-cloud/issues/363

SuaYoo committed:
fix checkbox

SuaYoo committed:
save browser profile and lang

SuaYoo committed:
don't save schedule if not cron type

SuaYoo committed:
fix back step

SuaYoo committed:
fix run now

SuaYoo committed:
fix lang type

Shrinks99 committed:
Updates language

Crawl job → Crawl

Some sections refer to it as "a crawl" because a crawl object is spawned from the crawl config.  Other sections refer to "the crawler", the program users are instructing.  I will document this language somewhere once I make a writing style guide for our documentation.

Shrinks99 commented:
Should we default to http if people enter a URL without the protocol?  Is this worth / does it require an error state?  Notably `example.com` is not flagged as invalid with a URL List crawl type.

<img width="546" alt="Screenshot 2022-12-01 at 3 41 40 PM" src="https://user-images.githubusercontent.com/5672810/205155491-eed13baa-f677-41b5-84d2-a45d2f076c54.png">

There are two "Crawl Limits" sections.  

<img width="895" alt="Screenshot 2022-12-01 at 6 01 33 PM" src="https://user-images.githubusercontent.com/5672810/205177219-10910ef9-a6b0-4610-b900-11edcabc752e.png">

### Info text Alignment

In the mockup I aligned this with the actual form control instead of the title of the control for hierarchy purposes.  I also recognize this isn't as easy as adding `pt-8` to them all because then they're out of wack with checkboxes that only have in-line labels.  Maybe we make a dupe `renderHelpTextColNoPadding` template without the `pt-8` class and use that one for form items without labels?  Wanted to run it by you here first.

## Naming Changes

After discussion with Ilya, we've agreed upon (and I think locked down) the name changes.  The previous change to "Crawl Jobs" has been axed in favour of sticking with "Crawls".  "Associated Jobs" also changed but that's not relevant here.  I have made a commit to this branch that should fix all instances of this in the UI however I haven't touched any variables or code changes.  Sorry this has happened kinda late in the game here :( 

Maybe consider changing the file name of `new-job-config.ts` as well but I'll leave that to you?

SuaYoo commented:
> There are two "Crawl Limits" sections.

There's a Crawl Limit and Crawl Job Limit section per Discord convos, it looks like the latter was renamed in your commit. What would be a better name for either section to differentiate them?

Re: Naming changes, they're mostly in https://github.com/webrecorder/browsertrix-cloud/pull/392, could you comment or make suggestions there @Shrinks99 ?

Shrinks99 commented:
Oof missed #392.  Revert my commit if need be!

SuaYoo committed:
rename type

SuaYoo committed:
rename crawl scale

SuaYoo committed:
rename images

SuaYoo committed:
rename crawl template

SuaYoo committed:
fix browser profile

SuaYoo committed:
Add confirmation step to New Job Config flow (#395)

SuaYoo committed:
update exclusion table

SuaYoo committed:
prevent overritingn lang

Shrinks99 commented:
Include linked pages is listed as `Include Linked Pages (“one hop out”)` in the confirm settings page for URL List crawls where the meaning is slightly different.

---

The "Browser Default" option is still present when users hit the clear field on the language selection dropdown which can't be selected otherwise in the list.  Is there a good utility reason for letting people use the browser's default instead of explicitly choosing a language option?

<img width="533" alt="Screenshot 2022-12-05 at 4 52 14 PM" src="https://user-images.githubusercontent.com/5672810/205750107-5f69c492-ec40-46e3-b60a-c633e942704e.png">

I've never been super happy with `clearable` on our dropdowns as it's placed right next to the indicator arrow and (IMO) makes the component's function more difficult to read & interact with.  I'd have fewer problems if the clear button was placed outside of the bounds of the dropdown button itself, but that's kinda on Shoelace and IMO out of scope for us to change.  In any case, for all our dropdowns I'm comfortable with forcing people to make an informed decision — especially for these instances where they're removing a default — and removing `clearable` in favour of a clearer dropdown button with less room for error when clicking because buttons within other buttons isn't really ideal, for text fields I'll give it a pass.

Default options or options different from all the other ones in the list like these can be put at the top with a separator instead if people want to remove them?  Thoughts appreciated if I've missed something

---

Language name & job name are set to "Not Specified" in the crawl summary when "Default Language" is selected for the browser language and the crawl seed URL is entered automatically for the crawl name.

<img width="893" alt="Screenshot 2022-12-05 at 5 15 06 PM" src="https://user-images.githubusercontent.com/5672810/205754361-78228b09-010e-4575-a4c6-483d284e5e3b.png">

SuaYoo committed:
update confirmation layout

SuaYoo committed:
fix job name not showing

SuaYoo commented:
> Include linked pages is listed as `Include Linked Pages (“one hop out”)` in the confirm settings page for URL List crawls where the meaning is slightly different.

Fixed!

> The "Browser Default" option is still present when users hit the clear field on the language selection dropdown which can't be selected otherwise in the list. Is there a good utility reason for letting people use the browser's default instead of explicitly choosing a language option?

Removed the clearable control so that language always defaults to the current user's browser language.

SuaYoo committed:
update exclusion table layout

SuaYoo committed:
show server error in confirmation page

SuaYoo committed:
remove unused code

SuaYoo committed:
switch to immutable merge

SuaYoo committed:
adjust for smaller screens

SuaYoo committed:
update form padding

Shrinks99 commented:
Re-aligned help text looks great! :D

Updated  by running Yarn, now the protocol being left off will result in an invalid field which is great.  After this is fixed the progress indicator icon didn't revert back to the editing state though :(

<img width="1158" alt="Screenshot 2022-12-06 at 3 41 47 PM" src="https://user-images.githubusercontent.com/5672810/206018554-27f05387-6871-4ff0-8001-0d2a5f8a4210.png">

Also the Crawl Time Limit field seems to accept non-numerical inputs.  Bonus points if these can auto-clear if they're entered and leave any remaining numbers: EG: if `te1st0` is entered, only `10` remains.  If no numbers remain the field is left blank and if the field is required (this one isn't) it results in an error.

This is also true for Page Time Limit.

Notably I assume the form actually does the validation because on the confirm screen these inputs are set as `None` but it would still be good logic to have to reduce mistakes & give feedback to users.

---
Browser Profile set to `Default` results in a `Not Specified` result on the final page.  Would be better to list it as `Default Profile` instead?  `Not Specified` is a little too close to being an error state IMO.

<img width="252" alt="Screenshot 2022-12-06 at 3 47 38 PM" src="https://user-images.githubusercontent.com/5672810/206019638-b96bf084-a2aa-4735-9f3b-37340589fa51.png">

---

At any stage of the config creation process where the user has entered data (all parts except the initial type selector page) we should warn them before leaving the page so their inputs aren't lost.  I think we currently use native dialogues for this in at least one other spot in the app today?  Would be good to do that here, especially because they can't go back in their browser to revisit the page with their inputs saved Google Form style.  Could spin this out into a separate PR.

---

The progress indicator bar is hidden at smaller breakpoints where it becomes horizontal.  Should be located above the steps.

<img width="1346" alt="Screenshot 2022-12-06 at 3 59 50 PM" src="https://user-images.githubusercontent.com/5672810/206022053-d6680253-5182-4d1d-b503-f388a3d27965.png">

SuaYoo committed:
prevent entering letters into fields

SuaYoo committed:
update browser profile default

SuaYoo committed:
rename file

SuaYoo committed:
update utc schedule with day

ikreymer committed:
backend: add 'jobType' field to top-level config object: 'url-list', 'seed-crawl', and a catch-all 'custom', fixes #363

SuaYoo committed:
update job type typing

SuaYoo committed:
save job type

SuaYoo committed:
duplicate job type

SuaYoo commented:
@ikreymer updated frontend to save job type. Duplicating a "custom" job type will show the "List of URLs" text box and scope type select with additional options: page, custom, and any.
<img width="527" alt="Screen Shot 2022-12-11 at 8 53 54 PM" src="https://user-images.githubusercontent.com/4672952/206963620-a3033cf0-4ca7-4619-8025-1099482a67fc.png">

SuaYoo committed:
handle custom type
